inc/Engine/Optimization/RUCSS/Admin/Subscriber.php

```diff
@@ -66,9 +75,44 @@ public static function get_subscribed_events() : array {
 			'admin_post_rocket_clear_usedcss'    => 'truncate_used_css_handler',
 			'admin_notices'                      => 'clear_usedcss_result',
 			'rocket_admin_bar_items'             => 'add_clean_used_css_menu_item',
+			'rocket_after_settings_checkbox'     => 'display_progress_bar',
+			'admin_enqueue_scripts'              => 'add_admin_js',
 		];
 	}
 
+	/**
+	 * Enqueue React params and Progress bar.
+	 *
+	 * @return void
+	 */
+	public function add_admin_js() {
+		// Retun on all pages but WP Rocket settings page.
+		$screen = get_current_screen();
+		if ( 'settings_page_wprocket' !== $screen->id ) {
+			return;
+		}
+
+		if ( ! $this->settings->is_enabled() ) {
+			return;
+		}
+
+		wp_enqueue_script( 'wpr-rucss-progress-bar', WP_ROCKET_ASSETS_JS_URL . 'react/rucss_progress_bar.js', [ 'react-dom' ], WP_ROCKET_VERSION, true );
+
+		$prewarmup_stats = $this->options_api->get( 'prewarmup_stats', [] );
+
+		wp_localize_script(
+			'wpr-rucss-progress-bar',
+			'rocket_rucss_ajax_data',
+			[
+				'api_url'                => rest_url( 'wp-rocket/v1/rucss/warmup/status' ),
+				'api_nonce'              => wp_create_nonce( 'rocket-ajax' ),
+				'api_debug'              => (bool) rocket_get_constant( 'WP_ROCKET_DEBUG' ),
+				'api_allow_optimization' => (bool) $prewarmup_stats['allow_optimization'],
```

Comment:
We should have an `isset()` check before trying to access the `allow_optimization` key